build.gradle

```diff
@@ -268,7 +268,7 @@ subprojects {
 // the dependency is added.
 gradle.projectsEvaluated {
   allprojects {
-    if (project.path == ':test:framework') {
+    if (project.path == ':test:framework' || project.path == ':client:test') {
       // :test:framework:test cannot run before and after :core:test
```

Comment:
Ok I see the confusion. The comment should be updated. It means that if the code below this were to run, a circular dependency would be added between `:test:framework:test` and `:core:test`. Likewise, the new check is added because `:client:test` would have a circular task dep with `:client:rest:test`.